From 026d6682815e09311a1b7b0410b56e5ccce5f705 Mon Sep 17 00:00:00 2001
From: Adriaan de Groot <groot@kde.org>
Date: Mon, 10 Sep 2018 13:52:50 +0200
Subject: [PATCH 1/3] (FreeBSD) Split implementation of VT switching

 - FreeBSD (obviously) doesn't have linux/vt.h, so it needs a
   different VT-switching implementation. Add a stub-file to
   do that -- the current implementation does nothing, which
   is what downstream patches effectively do by commenting
   out the relevant calls.
---
 src/daemon/CMakeLists.txt              |  9 ++++++-
 src/daemon/VirtualTerminal_FreeBSD.cpp | 37 ++++++++++++++++++++++++++
 2 files changed, 45 insertions(+), 1 deletion(-)
 create mode 100644 src/daemon/VirtualTerminal_FreeBSD.cpp

diff --git a/src/daemon/CMakeLists.txt b/src/daemon/CMakeLists.txt
index 09abc79..a175176 100644
--- a/src/daemon/CMakeLists.txt
+++ b/src/daemon/CMakeLists.txt
@@ -31,9 +31,16 @@ set(DAEMON_SOURCES
     SeatManager.cpp
     SignalHandler.cpp
     SocketServer.cpp
-    VirtualTerminal.cpp
 )
 
+# Different implementations of the VT switching code
+# (where the FreeBSD version does nothing).
+if(${CMAKE_SYSTEM} MATCHES "FreeBSD")
+    list(APPEND DAEMON_SOURCES VirtualTerminal_FreeBSD.cpp)
+else()
+    list(APPEND DAEMON_SOURCES VirtualTerminal.cpp)
+endif()
+
 qt5_add_dbus_adaptor(DAEMON_SOURCES "${CMAKE_SOURCE_DIR}/data/interfaces/org.freedesktop.DisplayManager.xml"          "DisplayManager.h" SDDM::DisplayManager)
 qt5_add_dbus_adaptor(DAEMON_SOURCES "${CMAKE_SOURCE_DIR}/data/interfaces/org.freedesktop.DisplayManager.Seat.xml"     "DisplayManager.h" SDDM::DisplayManagerSeat)
 qt5_add_dbus_adaptor(DAEMON_SOURCES "${CMAKE_SOURCE_DIR}/data/interfaces/org.freedesktop.DisplayManager.Session.xml"  "DisplayManager.h" SDDM::DisplayManagerSession)
diff --git a/src/daemon/VirtualTerminal_FreeBSD.cpp b/src/daemon/VirtualTerminal_FreeBSD.cpp
new file mode 100644
index 0000000..aaf1b61
--- /dev/null
+++ b/src/daemon/VirtualTerminal_FreeBSD.cpp
@@ -0,0 +1,37 @@
+/***************************************************************************
+* Copyright (c) 2015 Pier Luigi Fiorini <pierluigi.fiorini@gmail.com>
+*
+* This program is free software; you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation; either version 2 of the License, or
+* (at your option) any later version.
+*
+* This program is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with this program; if not, write to the
+* Free Software Foundation, Inc.,
+* 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+***************************************************************************/
+
+#include <QDebug>
+#include <QString>
+
+#include "VirtualTerminal.h"
+
+
+namespace SDDM {
+    namespace VirtualTerminal {
+        int setUpNewVt() {
+            qDebug() << "New VT is unsupported on FreeBSD";
+            return -1;
+        }
+
+        void jumpToVt(int vt, bool vt_auto) {
+            qDebug() << "Jumping to VT" << vt << "is unsupported on FreeBSD";
+        }
+    }
+}
-- 
2.25.1

